### PR TITLE
CATROID-691: user language sensor for locale

### PR DESCRIPTION
**Pull request by ndrnour**
* add new languageTags to list of languages {"af", "bn", "fi", "ha", "ig", "kk", "si", "sr", "tw"}
* refactor language setting

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [ ] Include the name of the Jira ticket in the PR’s title
- [ ] Include a summary of the changes plus the relevant context
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no compiler or linter warnings
- [ ] Perform a self-review of the changes
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s gitflow workflow
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/common/SharedPreferenceKeys.java [hunk 1]
```diff
@@ -34,11 +34,12 @@ private SharedPreferenceKeys() {
...
-			"sr-rSP", "da", "de", "en-rAU", "en-rCA", "en-rGB", "en", "es", "el", "fr", "gl", "hr", "it",
-			"sw", "hu", "mk", "ms", "nl", "no", "pl", "pt-rBR", "pt", "ru", "ro", "sq", "sl", "sk",
-			"sv", "vi", "tr", "uk", "bg", "ml", "ta", "kn", "te", "th", "gu", "hi", "ja", "ko", "lt", "zh-rCN",
-			"zh-rTW", "ar", "ur", "fa", "ps", "sd", "iw"};
+	public static final String[] LANGUAGE_TAGS = {DEVICE_LANGUAGE, "af", "az", "ms", "bs", "ca",
+			"da", "de", "en-AU", "en-CA", "en-GB", "en", "es", "fr", "gl", "ha", "hr", "ig", "id",
+			"it", "sw", "lt", "hu", "nl", "no", "pl", "pt-BR", "pt", "ro", "sq", "sk", "sl", "fi",
+			"sv", "vi", "tw", "tr", "cs", "el", "bg", "mk", "ru", "sr-CS", "sr", "sr-SP", "uk",
+			"kk", "he", "ur", "ar", "sd", "fa", "ps", "hi", "bn", "gu", "ta", "te", "kn", "ml",
+			"si", "th", "zh-CN", "zh-TW", "ja", "ko"};
```
[the remark this answers is not in the document]
Comment (ndrnour) [reply catroid/src/main/java/org/catrobat/catroid/common/SharedPreferenceKeys.java:42]: 'in' replaced by 'id' [Bahasa Indonesia](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes#ID)
'iw' replaced by 'he' [עברית (Hebrew)](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes#HE)

[the remark this answers is not in the document]
Comment (juliajulie95) [reply catroid/src/main/java/org/catrobat/catroid/common/SharedPreferenceKeys.java:42]: Thank you for clearing that up!

**Review by juliajulie95 — APPROVED** [review]

**Review by wslany — CHANGES_REQUESTED** [review]
Thanks! A few issues:

- Changing the language via the Catroid settings dialog is only reflected in a change in the sensor value _after_ the app has been force closed. A simple exit via back button from the app and "restart" is insufficient. Please correct this so that a change in the user interface language will be immediately reflected in the sensor's value when the project is executed the next time. I.e., no force closing should be necessary. Please add a test case for this.
- In the specification of the ticket, it was written that the response shall be as in https://jira.catrob.at/browse/CATROID-682 where it was defined as the language locale code, e.g., de_AT for Austrian German. However, currently the reported value of the string is "Deutsch (Österreich)" instead. Please correct this. The current way (full language name in the local language) makes it difficult for a project to handle these values since they depend on the set language. 
- The Catrobat language version number needs to be increased since a new sensor function has become available with this ticket.

**Comment by ndrnour** [comment]
> Thanks! A few issues:
> 
> * Changing the language via the Catroid settings dialog is only reflected in a change in the sensor value _after_ the app has been force closed. A simple exit via back button from the app and "restart" is insufficient. Please correct this so that a change in the user interface language will be immediately reflected in the sensor's value when the project is executed the next time. I.e., no force closing should be necessary. Please add a test case for this.
> * In the specification of the ticket, it was written that the response shall be as in https://jira.catrob.at/browse/CATROID-682 where it was defined as the language locale code, e.g., de_AT for Austrian German. However, currently the reported value of the string is "Deutsch (Österreich)" instead. Please correct this. The current way (full language name in the local language) makes it difficult for a project to handle these values since they depend on the set language.
> * The Catrobat language version number needs to be increased since a new sensor function has become available with this ticket.

fixed

**Review by wslany — APPROVED** [review]

**Comment by wslany** [comment]
Thanks!

**Pull request merged**